### PR TITLE
feat(framework) Add `node-config` arg to SuperNode

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/client/app.py [hunk 1]
```diff
@@ -204,6 +205,9 @@ def _start_client_internal(
         The IPv4 or IPv6 address of the server. If the Flower
         server runs on the same machine on port 8080, then `server_address`
         would be `"[::]:8080"`.
+    node_config: Dict[str, str]
+        The configuration of the node, it must contain at least a `num-partitions`
+        and a `partition-id` key.
```
Comment (danieljanes) [inline src/py/flwr/client/app.py:210]: This is not true - there is not requirement for `node_config` to contain specific keys

File: src/py/flwr/client/supernode/app.py [hunk 2]
```diff
@@ -67,7 +72,7 @@ def run_supernode() -> None:
         authentication_keys=authentication_keys,
         max_retries=args.max_retries,
         max_wait_time=args.max_wait_time,
-        partition_id=args.partition_id,
+        node_config=parse_config_args(args.node_config),
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:75]: We should change the argument names in `parse_config_args` to reflect this use case as well

File: src/py/flwr/client/supernode/app.py [hunk 3]
```diff
@@ -389,11 +395,11 @@ def _parse_args_common(parser: argparse.ArgumentParser) -> None:
...
     parser.add_argument(
-        "--partition-id",
+        "--node-config",
         type=int,
-        help="The data partition index associated with this SuperNode. Better suited "
-        "for prototyping purposes where a SuperNode might only load a fraction of an "
-        "artificially partitioned dataset (e.g. using `flwr-datasets`)",
+        help="A coma separated list of key/value pairs (separated by `=`) to configure "
+        "the SuperNode. "
+        "E.g, `--node-config key1='value1',partition-id=0,num-partitions=100`",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:402]:
```suggestion
        help="A comma separated list of key/value pairs (separated by `=`) to configure "
        "the SuperNode. "
        "E.g, `--node-config key1='value1',partition-id=0,num-partitions=100`",
```

File: src/py/flwr/client/supernode/app.py [hunk 4]
```diff
@@ -389,11 +395,11 @@ def _parse_args_common(parser: argparse.ArgumentParser) -> None:
...
     parser.add_argument(
-        "--partition-id",
+        "--node-config",
         type=int,
-        help="The data partition index associated with this SuperNode. Better suited "
-        "for prototyping purposes where a SuperNode might only load a fraction of an "
-        "artificially partitioned dataset (e.g. using `flwr-datasets`)",
+        help="A comma separated list of key/value pairs (separated by `=`) to configure "
+        "the SuperNode. "
+        "E.g, `--node-config key1='value1',partition-id=0,num-partitions=100`",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:402]:
```suggestion
        help="A comma separated list of key/value pairs (separated by `=`) to configure "
        "the SuperNode. "
        "E.g, `--node-config key1=\"value1\",partition-id=0,num-partitions=100`",
```

File: src/py/flwr/client/supernode/app.py (hunk 2 above)
Comment (jafermarq) [reply src/py/flwr/client/supernode/app.py:75]: changed input arg of `parse_config_args` from `config_override` to simply `config`.

File: src/py/flwr/client/app.py (hunk 1 above)
Comment (jafermarq) [reply src/py/flwr/client/app.py]: I've removed that sentence entirely.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**